numpy/core/tests/test_numeric.py

```diff
@@ -3511,6 +3511,12 @@ def test_broadcast_error_kwargs(self):
 
         assert_raises(ValueError, np.broadcast, 1, **{'x': 1})
 
+    def test_shape_mismatch_error_message(self):
+        with pytest.raises(ValueError, match=r"arg 0 with shape \(1, 3\) and "
+                                             r"arg 2 with shape \(2,\)"):
```

Comment:
Might be nicer to write `match = ...` to avoid the split within the string.  But doesn't matter.